### PR TITLE
MM-20003 Fix Fluentd issues with new chart

### PR DESCRIPTION
**Pull request by stylianosrigas**
Add ability to:
- Add new Helm repos
- Update Helm repos everytime to get latest available charts
- Install new Fluentd chart.

### CODE REVIEW

File: internal/provisioner/kops_provisioner_cluster.go [hunk 1]
```diff
@@ -213,6 +213,12 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, aws aw
 		return err
 	}
 
+	logger.Info("Adding https://kiwigrid.github.io Helm repo.")
+	addHelmRepo("kiwigrid", "https://kiwigrid.github.io", logger)
```
Comment (gabrieljackson) [inline internal/provisioner/kops_provisioner_cluster.go]: We are ignoring the returned error here. We should check it.

File: internal/provisioner/kops_provisioner_cluster.go [hunk 2]
```diff
@@ -213,6 +213,12 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, aws aw
...
+
+	logger.Info("Updating all Helm repos.")
+	helmRepoUpdate(logger)
```
Comment (gabrieljackson) [inline internal/provisioner/kops_provisioner_cluster.go]: Same here.

File: internal/provisioner/kops_utils.go [hunk 3]
```diff
@@ -112,6 +112,44 @@ func getLoadBalancerEndpoint(ctx context.Context, namespace string, logger log.F
...
+
+	cmd = exec.Command("helm repo add", arguments...)
+
+	logger.WithFields(log.Fields{
+		"cmd":  cmd.Path,
+		"args": cmd.Args,
+	}).Info("Invoking command")
+
+	err := cmd.Run()
+	if err != nil {
```
Comment (gabrieljackson) [inline internal/provisioner/kops_utils.go]: We could just return err here without checking it. It makes sense to keep the if check if you want to wrap the error with more info.

File: internal/provisioner/kops_utils.go [hunk 4]
```diff
@@ -112,6 +112,44 @@ func getLoadBalancerEndpoint(ctx context.Context, namespace string, logger log.F
...
+func helmRepoUpdate(logger log.FieldLogger) error {
+	var cmd *exec.Cmd
+	cmd = exec.Command("helm repo update")
+
+	logger.WithFields(log.Fields{
+		"cmd": cmd.Path,
+	}).Info("Invoking command")
+
+	err := cmd.Run()
+	if err != nil {
```
Comment (gabrieljackson) [inline internal/provisioner/kops_utils.go]: Same here.

File: internal/provisioner/kops_provisioner_cluster.go (hunk 1 above)
Comment (stylianosrigas) [reply internal/provisioner/kops_provisioner_cluster.go]: Thanks @gabrieljackson fixed :)

File: internal/provisioner/kops_utils.go (hunk 3 above)
Comment (stylianosrigas) [reply internal/provisioner/kops_utils.go]: Fixed ;)

File: internal/provisioner/kops_utils.go (hunk 4 above)
Comment (stylianosrigas) [reply internal/provisioner/kops_utils.go]: Fixed ;)

File: internal/provisioner/kops_provisioner_cluster.go (hunk 2 above)
Comment (stylianosrigas) [reply internal/provisioner/kops_provisioner_cluster.go]: Fixed ;)

**Pull request merged**